src/hyperlink/test/test_url.py

```diff
@@ -1116,6 +1138,18 @@ def test_autorooted(self):
         self.assertEqual(normal_absolute.rooted, True)
         self.assertEqual(attempt_unrooted_absolute.rooted, True)
 
+    def test_rooted_with_empty_non_none_host(self):
+        # type: () -> None
+        """
+        The C{rooted} constructor argument will be ignored on URLs that include
+        a scheme.
```

Comment:
This docstring doesn't seem to match with the test, as the ctor arg isn't involved. Maybe test that all four of these match `URL.from_text('udp://:4900')`? :

```
URL(scheme='udp', port=4900)
URL(scheme='udp', port=4900, rooted=False)
URL(scheme='udp', host='', port=4900)
URL(scheme='udp', host='', port=4900, rooted=False)
```

Reply:
Updated to more closely match the desired behavior that is being tested.

Reply:
Thanks for the review!